xds/server.go

```diff
@@ -75,30 +74,12 @@ type grpcServerInterface interface {
 	GracefulStop()
 }
 
-// ServeOptions contains parameters to configure the Serve() method.
-//
-// Experimental
-//
-// Notice: This type is EXPERIMENTAL and may be changed or removed in a
-// later release.
-type ServeOptions struct {
-	// Address contains the local address to listen on. This should be of the
-	// form "host:port", where the host must be a literal IP address, and port
-	// must be a literal port number. If the host is a literal IPv6 address it
-	// must be enclosed in square brackets, as in "[2001:db8::1]:80.
-	Address string
-}
-
-func (so *ServeOptions) validate() error {
-	addr, port, err := net.SplitHostPort(so.Address)
-	if err != nil {
-		return fmt.Errorf("xds: unsupported address %q for server listener", so.Address)
+func validateListener(lis net.Listener) error {
```

Comment:
Should this be returning the port? Don't we need that?

Reply:
Since we have to pass the `lis` to the `newListenerWrapper` function, we can get the address with another call to `lis.Addr().String()`.

And now that we are not checking if this is a `net.TCPListener`, I got rid of this function totally and inlined it into `Serve()`.

Reply:
Oh, I see -- we use the whole address and not just the port.